src/bin/Interpret.cc

```diff
@@ -971,6 +972,13 @@ bool Interpret::defineFun(const ASTNode& n)
     return rval;
 }
 
+bool Interpret::storeDefinedFun(std::string const & fname, const vec<PTRef> & args, SRef ret_sort, const PTRef tr) {
+    if (defined_functions.has(fname)) { return false; }
+
+    defined_functions.insert(fname, TemplateFunction(fname, args, ret_sort, tr), true); // TODO: ask config if defined are scoped
```

Comment:
The standard introduced option `:global-declarations`, but I don't think we have it yet. We need to add it to the config.